pkg/storage/parallel.go

```diff
@@ -4,25 +4,31 @@ import (
 	"context"
 	"errors"
 
-	"github.com/bacalhau-project/bacalhau/pkg/model"
-	"github.com/bacalhau-project/bacalhau/pkg/util/generic"
 	"github.com/rs/zerolog/log"
 	"go.ptx.dk/multierrgroup"
+
+	"github.com/bacalhau-project/bacalhau/pkg/model"
 )
 
+type PreparedStorage struct {
+	Spec   model.StorageSpec
+	Volume StorageVolume
+}
+
 // ParallelPrepareStorage downloads all of the data necessary for the passed
 // storage specs in parallel, and returns a map of specs to their download
 // volume counterparts.
 func ParallelPrepareStorage(
 	ctx context.Context,
 	provider StorageProvider,
-	specs []model.StorageSpec,
-) (map[*model.StorageSpec]StorageVolume, error) {
-	volumes := generic.SyncMap[*model.StorageSpec, StorageVolume]{}
+	specs ...model.StorageSpec,
+) ([]PreparedStorage, error) {
```

Comment:
Returns a slice now as the returned map keyed with pointers cannot cross an RPC boundary.